tests/PayloadTest.php

```diff
@@ -86,12 +86,38 @@ public function testConvertToJSon()
             ->setThreadId('tread-id')
             ->setContentAvailability(true)
             ->setMutableContent(true)
-            ->setCustomValue('key', 'value');
+            ->setCustomValue('key', 'value')
+            ->setCustomValue('custom_array', $custom_values);
+        $this->assertJsonStringEqualsJsonString(
+            '{"aps": {"alert": {"title": "title"}, "badge": 1, "sound": "sound", "category": "category", ' .
+            ' "thread-id": "tread-id", "mutable-content": 1, "content-available": 1}, "key": "value", "custom_array": [1,2,3]}',
+            $payload->toJson()
+        );
+        $decoded_json = json_decode($payload->toJson());
+        $this->assertEquals(gettype($decoded_json->custom_array), 'array');
+    }
 
+    public function testConvertToJSonWithObjectForce()
+    {
+        $custom_values = [1, 2, 3];
+        $alert = Alert::create()->setTitle('title');
+        $payload = Payload::create()
+            ->setAlert($alert)
+            ->setBadge(1)
+            ->setSound('sound')
+            ->setCategory('category')
+            ->setThreadId('tread-id')
+            ->setContentAvailability(true)
+            ->setMutableContent(true)
+            ->setCustomValue('key', 'value')
+            ->setCustomValue('custom_array', $custom_values)
+            ->setJSONOptions(JSON_UNESCAPED_UNICODE | JSON_FORCE_OBJECT);
         $this->assertJsonStringEqualsJsonString(
             '{"aps": {"alert": {"title": "title"}, "badge": 1, "sound": "sound", "category": "category", ' .
-            ' "thread-id": "tread-id", "mutable-content": 1, "content-available": 1}, "key": "value"}',
+            ' "thread-id": "tread-id", "mutable-content": 1, "content-available": 1}, "key": "value", "custom_array": {"0": 1, "1": 2, "2": 3}}',
```

Comment:
I can't imagine a case when we need such `custom_array` object. Maybe it is better to have only one way to get `custom_array` as array.

Reply:
Fixed by my new commit.